doc/mgmt/SONiC_OC_FDB_get_support_HLD.md

```diff
@@ -0,0 +1,271 @@
+
+# FDB get support
+
+Implement get support for FDB using CLI/REST/gNMI SONiC management framework interfaces.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/10/2019  |  Venkatesan Mahalingam      | Initial version                   |
+
+# About this Manual
+This document provides general information about FDB management get interfaces in SONiC.
+# Scope
+Covers Northbound get interface for the FDB feature, as well as Unit Test cases.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| FDB                      | Forwarding Database
+
+
+# 1 Feature Overview
+
+The documents covers the FDB get interfaces (CLI/REST/gNMI) for fetching the data from the back-end to the user using SONiC managment framework.
+
+## 1.1 Requirements
+
+![](http://10.59.132.240:9009/projects/csg_sonic/documentation/graphics/templates/test1.png)
+
+### 1.1.1 Functional Requirements
+
+Provide management framework get support to existing SONiC capabilities with respect to FDB.
+
+### 1.1.2 Configuration and Management Requirements
+ - Implement FDB CLI show Commands
+ - REST get support for FDB
+ - gNMI get support for FDB
+
+### 1.1.3 Scalability Requirements
+### 1.1.4 Warm Boot Requirements
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+### 1.2.2 Container
+### 1.2.3 SAI Overview
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+## 2.2 Functional Description
+# 3 Design
+## 3.1 Overview
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+## 3.4 SyncD
+## 3.5 SAI
+## 3.6 User Interface
+### 3.6.1 Data Models
+The following open config YANG model is used to implement get support for FDB entries.
+[https://github.com/openconfig/public/blob/master/release/models/network-instance/openconfig-network-instance-l2.yang#L292)
+```
```

Comment:
Are all of these attributes supported? If not, you should highlight the supported ones in `diff` format. If they are all supported, you should mention that just ahead of this code block.

Reply:
Done